package-res/resources/web/pentaho/type/_type.js

```diff
@@ -0,0 +1,1362 @@
+        return !!superType && (superType === this || O_isProtoOf.call(superType, this));
+      },
+
+      // TODO: is conversion always successful? If so, should it be?
+      /**
+       * Converts a value to an instance of this type,
+       * if it is not one already.
+       *
+       * If a {@link Nully} value is specified, `null` is returned.
+       *
+       * Otherwise, if a given value is not already an instance of this type
+       * (checked using [is]{@link pentaho.type.Type#is}),
+       * this method delegates the creation of an instance to
+       * [create]{@link pentaho.type.Type#create}.
+       *
+       * @param {?any} value The value to convert.
+       * @return {?pentaho.type.Instance} The converted value or `null`.
+       */
+      to: function(value) {
+        return value == null   ? null  :
+               this.is(value)  ? value :
+               this.create(value);
+      },
+
+      //region serialization
+      /**
+       * Creates a specification that describes this type.
+       *
+       * If an [ambient specification context]{@link pentaho.type.SpecificationContext.current},
+       * currently exists, it is used to manage the serialization process.
+       * Otherwise, one is created and set as current.
+       *
+       * This method creates a new {@link pentaho.type.SpecificationScope} for describing
+       * this type, and any other instances and types it references,
+       * and then delegates the actual work to {@link pentaho.type.Type#toSpecInContext}.
+       *
+       * @param {Object} [keyArgs] - The keyword arguments object.
+       * Passed to every type and instance serialized within this scope.
+       *
+       * Please see the documentation of subclasses for information on additional, supported keyword arguments.
+       *
+       * @param {?boolean} [keyArgs.isJson=false] - Generates a JSON-compatible specification.
+       * Attributes which don't have a JSON-compatible specification are omitted.
+       *
+       * @return {!pentaho.type.spec.ITypeProto} A specification of this type.
+       *
+       * @see pentaho.type.Type#toSpecInContext
+       * @see pentaho.type.Type#_fillSpecInContext
+       */
+      toSpec: function(keyArgs) {
+        return O.using(new SpecificationScope(), this.toSpecInContext.bind(this, keyArgs || {}));
+      },
+
+      /**
+       * Creates a specification that describes this type.
+       *
+       * This method requires that there currently exists an
+       * [ambient specification context]{@link pentaho.type.SpecificationContext.current}.
+       *
+       * The default implementation returns a plain object with the id of the type and
+       * any other specified attributes
+       * (like [label]{@link pentaho.type.Type#label} or [description]{@link pentaho.type.Type#description}).
+       *
+       * @param {Object} [keyArgs] - The keyword arguments object.
+       * Passed to every type and instance serialized within this scope.
```

Comment:
Suggest rewording into a sentence structure:

The keyword arguments object is passed to every type and instance serialized within this scope.